plugins/modules/networkpool.py

```diff
@@ -762,59 +784,60 @@ def validate_input(self, pool_params):
         if pool_params.get('additional_pool_params'):
             self.validate_additional_pool_params(pool_params)
 
-    def perform_module_operation(self):
-        """
-        Perform different actions based on parameters chosen in playbook
-        """
-        result = dict(
-            changed=False,
-            network_pool=[]
-        )
-        groupnet_name = self.module.params['groupnet_name']
-        subnet_name = self.module.params['subnet_name']
-        pool_name = self.module.params['pool_name']
-        state = self.module.params['state']
-        pool_params = self.module.params
-        modify_pool_param = None
-        new_pool_name = self.module.params['new_pool_name']
-
-        self.validate_input(pool_params)
-
-        input_param = {}
-        for param in self.module.params:
-            input_param[param] = self.module.params[param]
-
-        pool_details = self.get_network_pool(
-            groupnet_name, subnet_name, pool_name)
-
-        if state == "absent" and pool_details:
-            result['changed'] = self.delete_network_pool(
-                groupnet_name, subnet_name, pool_name)
-            result['network_pool'] = []
-
-        if state == "present" and not pool_details:
-            pool_params = self.construct_pool_parameters(input_param)
-            if pool_params:
-                self.create_network_pool(groupnet_name, subnet_name, pool_params)
-                result['changed'] = True
-
-        # Form dictionary of modifiable parameters for a pool
-        if pool_details and input_param:
-            modify_pool_param = \
-                self.is_pool_modifiable(pool_details, pool_params)
-
-        # Modify network pool
-        if modify_pool_param and state == "present":
-            self.modify_network_pool(modify_pool_param, pool_name, groupnet_name, subnet_name)
-            if new_pool_name:
-                pool_name = new_pool_name
-            result['changed'] = result['modify_network_pool'] = True
-
-        if state == 'present':
-            result['network_pool'] = \
-                self.get_network_pool(groupnet_name, subnet_name, pool_name)
-
-        self.module.exit_json(**result)
+    def get_diff_after(self, pool_params, network_pool_details):
+        """Get diff between playbook input and network pool details
+        :param pool_params: Dictionary of parameters input from playbook
+        :param network_pool_details: Dictionary of network pool details
+        :return: Dictionary of parameters of differences"""
+
+        if pool_params["state"] == "absent":
+            return {}, {}
+        else:
+            modify_params = {}
+            if network_pool_details is None:
+                diff_dict = {
```

Comment:
There are some diff values are hardcoded, will it not be different? is it okay to fetch from arrays and add it

Reply:
They are just the expected output, these cannot be modified.